### ISSUE TITLE
when sync a commit it reverts neoforge.tmol

### ISSUE THREAD

**Issue by Thunderrock424242** [Defeatomizer/MCreatorVCSPlugin#35]
### Issue description

the mods file if you have a user code block it completely reverts it to no code block

### How to reproduce this issue?

1. write soemthing in the code block
2. sync commit and watch the magic

### Operating system

Windows

### Details

mcreator 2024.2

### Remote repository

![Screenshot 2024-07-18 111445](https://github.com/user-attachments/assets/b29e1352-276d-4631-a6ac-84ae87fd397a)

**Comment by Thunderrock424242** [comment]
It's very annoying as I have to change it constantly when I push a update to GitHub since I use mixins

**Comment by Defeatomizer** [comment]
Are you modifying the user code block on a remote branch?

**Comment by Defeatomizer** [comment]
Are you only modifying the user code block on a remote branch or on a local one?

**Comment by Thunderrock424242** [comment]
> Are you only modifying the user code block on a remote branch or on a local one? 

Yes In the remote branch and when I merge it it does 2 commits to fix my mods.tmol file

**Comment by Thunderrock424242** [comment]
It also changes it inside mcreator

**Comment by Defeatomizer** [comment]
Did you try editing it on the local branch instead (directly from MCreator)?

**Comment by Thunderrock424242** [comment]
I have not tried to edit it on GitHub I will let you know what happens

**Comment by Defeatomizer** [comment]
The plugin regenerates base mod files before synchronizing remote and local branches, and one of latest MCreator versions was reported to clear contents of user code blocks when regenerating code of the workspace. Not sure if these two are related because I haven't used user code blocks yet, so if you haven't installed the latest MCreator release where this bug is fixed (2024.2.28714), I recommend to try installing it and changing the base mod file from it.

**Comment by Defeatomizer** [comment]
The plugin regenerates base mod files before synchronizing local and remote branches, and one of latest MCreator versions was reported to clear contents of user code blocks when regenerating code of the workspace. Not sure if these two are related because I haven't used user code blocks yet, so if you haven't installed the latest MCreator release where this bug is fixed (2024.2.28714), I recommend to try installing it and changing the base mod file from it.

**Comment by Thunderrock424242** [comment]
I am using the latest release

**Comment by Thunderrock424242** [comment]
> The plugin regenerates base mod files before synchronizing local and remote branches, and one of latest MCreator versions was reported to clear contents of user code blocks when regenerating code of the workspace. Not sure if these two are related because I haven't used user code blocks yet, so if you haven't installed the latest MCreator release where this bug is fixed (2024.2.28714), I recommend to try installing it and changing the base mod file from it.

Tested it and it still reverts the code even if I do the code in the GitHub it reverts it after the commit

**Comment by Thunderrock424242** [comment]
It looks like it also changes how mcreator regens code as you see in the neoforgemods.tmol you can see the changes it makes

**Comment by Defeatomizer** [comment]
> even if I do the code in the GitHub

The same happens after editing the same file via MCreator's built-in code editor?

**Comment by Defeatomizer** [comment]
> even if I do the code in the GitHub

The same happens after editing the same file via MCreator's built-in code editor? This is still not clean for me from your replies.

**Comment by Thunderrock424242** [comment]
This only seems to be happening in the mods.tmol so there's no code editor for it I just open the file and edit it from there

**Comment by Defeatomizer** [comment]
MCreator's code editor supports TOML files, just locate it in the workspace file browser and open it.

**Comment by Thunderrock424242** [comment]
Yes Ik that's what I am have been editing for my user code blocks

**Comment by Thunderrock424242** [comment]
Would it be easier if I just make a video and show you??

**Comment by Defeatomizer** [comment]
> Yes Ik that's what I am have been editing for my user code blocks

Hmm, I'll see what can be done.

**Comment by Thunderrock424242** [comment]
You can check out the GitHub repository I sent in my issue report so you can see what's being reverted

**Comment by Defeatomizer** [comment]
Does the same happen to other user code blocks (e.g. in main mod class)?

**Comment by Thunderrock424242** [comment]
> Does the same happen to other user code blocks (e.g. in main mod class)?

No only the one I mentioned

**Comment by Defeatomizer** [comment]
Can you please make a commit (maybe in a temporary branch) where you modify some another one locally and send it to remote?

**Comment by Thunderrock424242** [comment]
> Can you please make a commit (maybe in a temporary branch) where you modify some another one locally and send it to remote?

Wdym can you explain how to do that?

**Comment by Defeatomizer** [comment]
In MCreator, add some dummy text (e.g. a comment, like `// Test data`) to one of the code blocks in `WildernessOddesyApiMod.java` and send these changes to your remote repository.

**Comment by Thunderrock424242** [comment]
Ok

**Comment by Thunderrock424242** [comment]
> In MCreator, add some dummy text (e.g. a comment, like `// Test data`) to one of the code blocks in `WildernessOddesyApiMod.java` and send these changes to your remote repository.

Done

**Comment by Defeatomizer** [comment]
So the other code block gets reverted too. Are you sure you're using the version I specified? Judging by the fact you also applied some formatting in both files, it looks like the base mod files get regenerated automatically and erase any modifications to them, including user code blocks, which should not be the case in that version.

**Comment by Defeatomizer** [comment]
If the version is right, this may only happen if things like the merge resolve sides (base, local and remote) are determined incorrectly, but I really doubt they are as my commits to a test repository worked as intended.
P. S. Can you attach the screenshot of the **About...** dialog?

**Comment by Thunderrock424242** [comment]
Which file is that and btw do you think you can fix it?

**Comment by Defeatomizer** [comment]
This plugin should not be able do apply such changes, the only reason to do that is if you're making a newer commit on the remote branch before pushing the local one, which does not seem to be your case.

**Comment by Defeatomizer** [comment]
> Can you attach the screenshot of the **About...** dialog?

**Comment by Thunderrock424242** [comment]
Where do I find the about dialog?

**Comment by Defeatomizer** [comment]
One of the ways is the main menu bar -> **Help**.

**Comment by Thunderrock424242** [comment]
![image](https://github.com/user-attachments/assets/f7a62ee1-1bdd-4fb8-9370-ae54695b1e89)
This?

**Comment by Defeatomizer** [comment]
The one at the bottom (**About MCreator**).

**Comment by Thunderrock424242** [comment]
What should I look for in it when I push it

**Comment by Defeatomizer** [comment]
Hmm, nevermind, the same version string is saved in your workspace file. It's quite weird that things like these happen even after being fixed in MCreator itself.

**Comment by Thunderrock424242** [comment]
> Hmm, nevermind, the same version string is saved in your workspace file. It's quite weird that things like these happen even after being fixed in MCreator itself.

You think you can fix it or is it not fixable.
